### ISSUE TITLE
[CT-1496] [Feature] BigQuery - Custom Dataproc Images for python models

### ISSUE THREAD

**Issue by roivant-matts** [dbt-labs/dbt-bigquery#384]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-bigquery functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Hello, I gave a quick read of the dbt python docs. For bigquery it seemed the execution options are serverless and cluster with cluster being more flexible in terms of dependencies. The GCP dataproc docs seem to indicate custom images can be used with the serverless option. I have a use case involving R (would use rpy2), so am thinking of an image with R installed, required packages and then required python packages. Should this be possible with dbt python models (e.g. in some config specify serverless and a custom image on GCP's image registry)? I'd rather avoid a cluster which is always running and would benefit from the serverless config.

**Comment by torkjel** [comment]
I believe #381 would support this. You'd just add this to your profile configuration:
```
      dataproc_batch:
        runtime_config:
          container_image_ <url to image>
          properties:
            <etc...>
```

**Comment by Fleid** [comment]
Resolved by #381
